### PR TITLE
feat: add new maps with case sensitive keys

### PR DESCRIPTION
**Pull request by fzipi**
## what

- add a new type of maps with case sensitive keys

## why

- all maps are case insensitive on keys now

### CODE REVIEW

File: internal/collections/map_test.go [hunk 1]
```diff
@@ -21,8 +21,53 @@ import (
...
+	}
+	if l := len(c.FindRegex(regexp.MustCompile("user.*"))); l != 2 {
+		t.Errorf("Error should find regex, got %d", l)
+	}
+
+	c.Add("user-agent", "value3")
+
+	wantStr := `REQUEST_HEADERS:
+    user: value
+    user-agent: value2,value3
```
Comment (jcchavezs) [inline internal/collections/map_test.go:47]: nit
```suggestion
    user-agent: value2; value3
```

Comment (jptosso) [reply internal/collections/map_test.go:47]: JC RFC 2616

> Multiple message-header fields with the same field-name MAY be present in a message if and only if the entire field-value for that header field is defined as a comma-separated list [i.e., #(values)]. It MUST be possible to combine the multiple header fields into one "field-name: field-value" pair, without changing the semantics of the message, by appending each subsequent field-value to the first, each separated by a comma. The order in which header fields with the same field-name are received is therefore significant to the interpretation of the combined field value, and thus a proxy MUST NOT change the order of these field values when a message is forwarded.

Comment (fzipi) [reply internal/collections/map_test.go:47]: @jcchavezs do you still think we need to change something here?

**Comment by fzipi** [comment]
I'll fix tests today.

File: internal/collections/map_test.go (hunk 1 above)
Comment (jcchavezs) [reply internal/collections/map_test.go:47]: no need

**Review by M4tteoP — COMMENTED** [review]
LGTM, what is the plan here? Merge this in Coraza 3.x and then hold https://github.com/corazawaf/coraza/pull/1042 (The breaking change) as part of [v4 milestones](https://github.com/corazawaf/coraza/issues/945)?

**Comment by fzipi** [comment]
I'll be pushing smaller changes until we figure it out. I don't think this will break the api for a major release though.

**Comment by fzipi** [comment]
Can someone from @corazawaf/core-developers approve this, so I can keep pushing stuff on top of this one?

**Review by jcchavezs — APPROVED** [review]

**Pull request merged**